lib/api/filter.rb

```diff
@@ -33,22 +33,19 @@ def initialize(filters, model)
     def parse
       filters.select(&:present?).each do |filter|
         parsed_filter = parse_filter(filter)
-        *associations, attr = parsed_filter[:attr].split(".")
-        if associations.size > 1
+        field = parsed_filter[:field]
+        if field.associations.size > 1
           raise BadRequestError, "Filtering of attributes with more than one association away is not supported"
         end
-        unless virtual_or_physical_attribute?(target_class(model, associations), attr)
-          raise BadRequestError, "attribute #{attr} does not exist"
+        unless field.valid?
+          raise BadRequestError, "attribute #{field} does not exist"
```

Comment:
Does `field.to_s` print the right thing here, or will be print out like `#<MiqExpression::FIeld @...>`